cpp/src/arrow/util/int128.h

```diff
@@ -101,7 +102,7 @@ class ARROW_EXPORT Int128 {
   uint64_t low_bits() const { return low_bits_; }
 
   /// \brief Put the raw bytes of the value into a pointer to uint8_t.
-  void ToBytes(uint8_t** out) const;
+  Status ToBytes(std::array<uint8_t, 16>* out) const;
```

Comment:
Having not used `std::array` much I couldn't say whether this is preferable to `std::string` for transporting bytes. Seems OK for the moment, though

Reply:
This is similar to what I was doing before I started using `std::array`. I was using raw fixed-length C arrays, which are more cumbersome to use with the STL and just as efficient.